### PR TITLE
FOGL-930 get interest endpoint added with interest registry tests

### PR DESCRIPTION
**Pull request by ashish-jabble**
- [x] GET /foglamp/interest endpoint added
- [x] test_interest_registry

**Tests O/p**

```
collected 17 items 

test_interest_registry.py::TestInterestRegistryApi::test_register_interest PASSED
test_interest_registry.py::TestInterestRegistryApi::test_unregister_interest PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_by_category PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_by_microservice_id PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_by_category_and_microservice_id PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_params_with_bad_data[-404-No interest registered] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_params_with_bad_data[?any=123-404-No interest registered] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_params_with_bad_data[?category=blah-404-No interest registered for category blah] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_params_with_bad_data[?microserviceid=foo-400-Invalid microservice id foo] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_params_with_bad_data[?microserviceid=d2abe6d7-ce77-448a-b13f-b2ada202b63b-404-No interest registered microservice id d2abe6d7-ce77-448a-b13f-b2ada202b63b] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_get_params_with_bad_data[?microserviceid=d2abe6d7-ce77-448a-b13f-b2ada202b63b&category=foo-404-No interest registered for category foo and microservice id d2abe6d7-ce77-448a-b13f-b2ada202b63b] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_register_with_bad_data[data0-400-Failed to register interest. microservice_uuid cannot be None] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_register_with_bad_data[data1-400-Invalid microservice id 0xe6ebd0] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_register_with_bad_data[data2-400-Failed to register interest. category_name cannot be None] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_unregister_with_bad_data[blah-400-Invalid registration id blah] PASSED
test_interest_registry.py::TestInterestRegistryApi::test_unregister_with_bad_data[d2abe6d7-ce77-448a-b13f-b2ada202b63b-404-InterestRecord with registration_id d2abe6d7-ce77-448a-b13f-b2ada202b63b does not exist] PASSED
==== 17 passed in 0.44 seconds ===
```
**Syslog**
```
Jan 11 16:46:51 nerd-034 [FOGLAMP] 01-11-2018 16:46:51 - INFO :: server: foglamp.services.core.server: start core
Jan 11 16:46:51 nerd-034 [FOGLAMP] 01-11-2018 16:46:51 - INFO :: server: foglamp.services.core.server: Management API started on http://0.0.0.0:42921
Jan 11 16:46:51 nerd-034 [FOGLAMP] 01-11-2018 16:46:51 - INFO :: server: foglamp.services.core.server: Announce management API service
Jan 11 16:46:51 nerd-034 [FOGLAMP] 01-11-2018 16:46:51 - INFO :: server: foglamp.services.core.server: start storage, from directory /home/foglamp/Development/FogLAMP/scripts
Jan 11 16:46:51 nerd-034 storage: Using default configuration:  { "plugin" : {  "value" : "postgres" }, "threads" : { "value" : "1" },  "port" : { "value" : "0" }, "managementPort" : { "value" : "0" } }.
Jan 11 16:46:51 nerd-034 port[20000]: Load storage plugin postgres.
Jan 11 16:46:51 nerd-034 port[20000]: Loaded storage plugin postgres.
Jan 11 16:46:51 nerd-034 port[20000]: Starting management api on port 0.
Jan 11 16:46:51 nerd-034 port[20000]: Starting service...
Jan 11 16:47:01 nerd-034 [FOGLAMP] 01-11-2018 16:47:01 - INFO :: service_registry: foglamp.services.core.service_registry.service_registry: Registered service instance id=2f525250-8a4c-451a-a337-182a1f6ff7f4: <FogLAMP Storage, type=Storage, protocol=http, address=localhost, service port=43661, management port=46653, status=1>
Jan 11 16:47:01 nerd-034 port[20000]: Registered service 2f525250-8a4c-451a-a337-182a1f6ff7f4.
Jan 11 16:47:01 nerd-034 port[20000]: Failed to parse result of category registration:
Jan 11 16:47:06 nerd-034 [FOGLAMP] 01-11-2018 16:47:06 - INFO :: server: foglamp.services.core.server: start scheduler
Jan 11 16:47:06 nerd-034 [FOGLAMP] 01-11-2018 16:47:06 - INFO :: scheduler: foglamp.services.core.scheduler.scheduler: Starting
Jan 11 16:47:06 nerd-034 [FOGLAMP] 01-11-2018 16:47:06 - INFO :: scheduler: foglamp.services.core.scheduler.scheduler: Starting Scheduler: Management port received is 42921
Jan 11 16:47:06 nerd-034 [FOGLAMP] 01-11-2018 16:47:06 - INFO :: scheduler: foglamp.services.core.scheduler.scheduler: Sleeping for 9999999 seconds
Jan 11 16:47:06 nerd-034 [FOGLAMP] 01-11-2018 16:47:06 - INFO :: server: foglamp.services.core.server: REST API Server started on http://0.0.0.0:8081
Jan 11 16:47:06 nerd-034 [FOGLAMP] 01-11-2018 16:47:06 - INFO :: service_registry: foglamp.services.core.service_registry.service_registry: Registered service instance id=da37e137-11f6-46ad-b59b-9995b3b5162b: <FogLAMP Core, type=Core, protocol=http, address=0.0.0.0, service port=8081, management port=42921, status=1>
Jan 11 16:47:48 nerd-034 [FOGLAMP] 01-11-2018 16:47:48 - INFO :: scheduler: foglamp.services.core.scheduler.scheduler: Processing stop request
Jan 11 16:47:48 nerd-034 [FOGLAMP] 01-11-2018 16:47:48 - INFO :: scheduler: foglamp.services.core.scheduler.scheduler: Stopped
Jan 11 16:47:48 nerd-034 port[20000]: Storage service shutdown in progress.
Jan 11 16:47:48 nerd-034 [FOGLAMP] 01-11-2018 16:47:48 - INFO :: service_registry: foglamp.services.core.service_registry.service_registry: Unregistered service instance id=2f525250-8a4c-451a-a337-182a1f6ff7f4: <FogLAMP Storage, type=Storage, protocol=http, address=localhost, service port=43661, management port=46653, status=1>
Jan 11 16:47:48 nerd-034 port[20000]: Unregistered service 2f525250-8a4c-451a-a337-182a1f6ff7f4.
Jan 11 16:47:48 nerd-034 port[20000]: Storage service shut down.
```

**Note:** Failed to parse result of category registration: --- failed due to invalid format of microservice uuid4. we need to fix it from storage end
`{'category': 'STORAGE', 'service': '0xe1ebd0'}`

### CODE REVIEW

File: tests/unit-tests/python/foglamp_test/services/core/interest_registry/test_interest_registry.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        assert 1 == len(interests)
+
+        assert data2["category"] == interests[0]["category"]
+        assert data2["service"] == interests[0]["microserviceId"]
+
+    @pytest.mark.parametrize("request_params, response_code, response_message", [
+        ('', 404, "No interest registered"),
+        ('?any=123', 404, "No interest registered"),
+        ('?category=blah', 404, "No interest registered for category blah"),
+        ('?microserviceid=foo', 400, "Invalid microservice id foo"),
```
Comment (praveen-garg) [inline tests/unit-tests/python/foglamp_test/services/core/interest_registry/test_interest_registry.py]: this should move to next test, BAD request

File: tests/unit-tests/python/foglamp_test/services/core/interest_registry/test_interest_registry.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        ('?microserviceid=d2abe6d7-ce77-448a-b13f-b2ada202b63b', 404,
+         'No interest registered microservice id d2abe6d7-ce77-448a-b13f-b2ada202b63b'),
+        ('?microserviceid=d2abe6d7-ce77-448a-b13f-b2ada202b63b&category=foo', 404,
+         'No interest registered for category foo and microservice id d2abe6d7-ce77-448a-b13f-b2ada202b63b')
+    ])
+    async def test_get_params_with_bad_data(self, request_params, response_code, response_message):
```
Comment (praveen-garg) [inline tests/unit-tests/python/foglamp_test/services/core/interest_registry/test_interest_registry.py]: better to make 2 tests 1 for 400 and 1 for 404

File: tests/unit-tests/python/foglamp_test/services/core/interest_registry/test_interest_registry.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        r = requests.get(BASE_URL + '/interest{}'.format(request_params))
+        assert response_code == r.status_code
+        assert response_message == r.reason
+
+    @pytest.mark.parametrize("registration_id, response_code, response_message", [
+        ('blah', 400, "InterestRecord with registration_id blah does not exist"),
+        ('d2abe6d7-ce77-448a-b13f-b2ada202b63b', 400,
+         "InterestRecord with registration_id d2abe6d7-ce77-448a-b13f-b2ada202b63b does not exist")
+    ])
```
Comment (praveen-garg) [inline tests/unit-tests/python/foglamp_test/services/core/interest_registry/test_interest_registry.py]: for these 2 records I believe it should be 404 | does not exist

**Comment by ashish-jabble** [comment]
@MarkRiddoch Also please see https://scaledb.atlassian.net/browse/FOGL-937

**Pull request merged**